packages/compile-solidity/src/compilerSupplier/index.ts

```diff
@@ -0,0 +1,117 @@
+import path from "path";
+import fs from "fs";
+import semver from "semver";
+
+import { Docker, Local, Native, VersionRange } from "./loadingStrategies";
+
+const defaultSolcVersion = "0.5.16";
+
+export class CompilerSupplier {
```

Comment:
@haltman-at suggested [this issue](https://github.com/trufflesuite/truffle/issues/4261), identifying that this CompilerSupplier implementation does too much manual retry logic, rather than using something like axios-retry.

Reply:
I posit that, since there's an open issue, this needn't block merge of this PR.